tools/runners/run-test-suite.sh

```diff
@@ -59,6 +59,25 @@ fi
 
 rm -f $TEST_FAILED $TEST_PASSED
 
+ROOT_DIR=""
+CURRENT_DIR=`pwd`
+PATH_STEP=2
+while true
+do
+    TMP_ROOT_DIR=`(echo $CURRENT_DIR; echo $0; echo $ENGINE; echo $TESTS) | cut -f1-$PATH_STEP -d/ | uniq -d`
```

Comment:
Please, explain. Which part, why, and do you have a fix? Did you run into a problem with it? (In the standard setup, i.e., when run by the makefile only, such a situation should not arise.)
PS: Please note that if this loop cannot find a common prefix then it will simply not remove anything from the printed paths. That's all to its effects. It does not change the actual execution at all.